hw/ip/spi_host/doc/_index.md

```diff
@@ -1288,7 +1288,168 @@ Highlights for this unit:
+   - For multi-segment commands, be sure to assert {{< regref "COMMAND.CSAAT" >}} for all but the last command segment
+7. For commands which expect to receive a reply, the data can then be read back from the {{< regref "DATA" >}} window.
+
+These latter four steps are then repeated for each command.
+Each step is covered in detail in the following sections.
+
+For concreteness, this Programmer's Guide uses examples from one of our primary target devices, the [W25Q01JV flash from Winbond](https://www.winbond.com/resource-files/W25Q01JV%20SPI%20RevB%2011132019.pdf).
+The SPI_HOST IP is however suitable for interacting with any number of SPI devices, and the same mode of operation can be used for any SPI device.
+
+## Initializing the IP
+
+### Per-target Configuration
+
+The {{< regref "CONFIGOPTS" >}} multi-register must be programmed to reflect the requirements of the attached target devices.
+As such these registers can be programmed once at initialization, or whenever a new device is connected (e.g., via changes in the external pin connections, or changes in the pinmux configuration).
+The proper settings for the {{< regref "CONFIGOPTS" >}} fields (e.g., CPOL and CPHA, clock divider, ratios, and other timing or sampling requirements) will all depend on the specific device attached as well as the board level delays.
+
+### Interrupt configuration
+
+The next step is to configuration the interrupts for the SPI_HOST.
+This should also be done at initialization using the following register fields:
+- The {{< regref "ERROR_ENABLE" >}} register should be configured to indicate what types of error conditions (if any) should be ignored to not trigger an interrupt.
+At reset, these fields are all high indicating that all error classes trigger an interrupt.
+
+- For interrupt driven I/O the {{< regref "EVENT_ENABLE" >}} register must be configured to select the desired event interrupts to signal the desired conditions (e.g. "FIFO empty", "FIFO at the watermark level", or "ready for next command segment").
+By default, this register is all zeros, meaning all event interrupts are disabled, and thus all transactions must be managed by polling the status register.
+   - When using the FIFO watermarks to send interrupts, the watermark levels must be set via the {{< regref "CONTROL.RX_WATERMARK" >}} and {{< regref "CONTROL.TX_WATERMARK" >}} fields.
+
+- The event and error interrupts must finally be enabled using the {{< regref "INTR_ENABLE" >}} register.
+
+### Enabling the SPI_HOST
+
+The IP must be enabled before sending the first command by asserting the {{< regref "CONTROL.SPIEN" >}} bit.
+
+## Issuing Commands
+
+As mentioned above, each command is typically specified in three phases: loading the TX data, specifying the command segments/format, and reading the RX data.
+In principle, the first two steps can be performed in either order.
+If the SPI_HOST FSM does not find any data to transmit it will simply stall until data is inserted.
```

Comment:
FSM could be omitted, or expanded to "state machine".

Reply:
```suggestion
If the SPI_HOST does not find any data to transmit it will simply stall until data is inserted.
```